lib/mayaUsd/render/vp2ShaderFragments/usdPreviewSurfaceCombiner.xml

```diff
@@ -87,12 +101,31 @@ usdPreviewSurfaceCombiner(
     result.outColor += diffuseColor * irradianceEnv;
 
     if (!mayaAlphaCut) {
-        result.outColor *= saturate(1.0 - transparency); 
+        result.outColor *= saturate(1.0 - transparency);
     }
 
     // Specular
     result.outColor += specularIrradianceIn;
-    result.outColor += specularColor * specularEnv;
+    if (useSpecularWorkflow) {
+        result.outColor += specularColor * specularEnv;
+    } else {
+        float R = (1.0 - ior) / (1.0 + ior);
+        float Rsq = R*R;
+
+        // Per https://graphics.pixar.com/usd/docs/UsdPreviewSurface-Proposal.html
+        //   If metallic is 1, then both F0 (reflectivity at 0 degree incidence) and edge F90 reflectivity will simply be the Albedo.
+        //   If metallic is 0, then Albedo is ignored in the calculation of F0 and F90; F0 is derived from ior via ( (1-ior)/(1+ior) )^2 and F90 is white. In between, we interpolate.
+
+        // So let's calculate F0 as a function of metalness...
+        float3 F0 = mix(float3(Rsq, Rsq, Rsq), specularEnv, metallic);
+        // ...and calculate F90 as a function of metalness....
+        float3 F90 = mix(float3(1.0, 1.0, 1.0), specularEnv, metallic);
+        // ... finally get the incidence angle to find the resulting outcolor
+        //     and multiply by the intensity of the env irradiance to make sure
+        //     we aren't lifting colors more than the lighting in the scene
+        float incidence = acos(dot(Lw, Nw))/90.0;
```

Comment:
Should be `SchlickFresnel(EdotH)`, but that would require functions and values that are given to  usdPreviewSurfaceLighting.xml but not to the combiner... I start to see how difficult this is going to be.

Reply:
> Should be SchlickFresnel(EdotH), but that would require functions and values that are given to usdPreviewSurfaceLighting.xml but not to the combiner... I start to see how difficult getting this perfectly aligned is going to be.

Hi:

There is this language specifically in the UsdPreviewSurface proposal that confuses me when thinking about how to lerp between F0/F90:

> useSpecularWorkflow = 1: (Specular workflow ) 
> specularColor - color3f - (0.0, 0.0, 0.0)
> Specular color to be used. This is the color at 0 incidence. Edge color is assumed white. Transition between the two colors according to Schlick fresnel approximation.
> useSpecularWorkflow = 0:   (Metalness workflow ) 
> metallic - float - 0.0
> Use 1 for metallic surfaces and 0 for non-metallic.
> - If metallic is 1, then both F0 (reflectivity at 0 degree incidence) and edge F90 reflectivity will simply be the Albedo.
> - If metallic is 0, then Albedo is ignored in the calculation of F0 and F90; F0 is derived from ior via ( (1-ior)/(1+ior) )^2 and F90 is white. *In between, we interpolate*.

It seems to imply that Schlick's fresnel should be used only when specular workflow is observed, and when in a metalness one, the word "interpolate" seemed ambiguous (I am not familiar with the math described in the metalness workflow), so we elected to try and follow the language (a.k.a. lerp between F0/F90 based on angle of incidence) in that case. Is this an incorrect assumption and we should still be using Schlick's fresnel for the interpolant term?

I found this quote from https://learnopengl.com/PBR/Theory which supports the statement above (i.e. use Schlick's fresnel for metalness too instead of using direct angle of incidence), but not sure if it applies.

> There are a few subtleties involved with the Fresnel equation. One is that the Fresnel-Schlick approximation is only really defined for dielectric or non-metal surfaces. For conductor surfaces (metals), calculating the base reflectivity with indices of refraction doesn't properly hold and we need to use a different Fresnel equation for conductors altogether. As this is inconvenient, we further approximate by pre-computing the surface's response at normal incidence (F0) at a 0 degree angle as if looking directly onto a surface. We interpolate this value based on the view angle, as per the Fresnel-Schlick approximation, such that we can use the same equation for both metals and non-metals.

Reply:
I'm trying to use the SchlickFresnel...but I'm stumped by `EdotH`. I don't see it declared as a property in `usdPreviewSurfaceLighting.xml` and I don't see it calculated anywhere.

I do however see a `VdotH` that is never used. My guess is that `EdotH` is a typo and it should be `VdotH` and the properties just get passed in positional order rather than any sort of name ? I just wanted to confirm though before making that assumption.

Reply:
Not a typo. If you look at the declaration section of the fragment you will see that the VdotH fragment input provided by Maya becomes the named parameter EdotH in the function call. Name mismatch are allowed as long as the parameter order and type matches.
V is for "view" and E is for "eye".

Reply:
Thanks for clarifying. That makes sense. I'll add a comment in the file for future clarity.

Reply:
Switched to an inline Schlick fresnel copy and it looks good to my eyes at least.